x-pack/plugins/logstash/server/routes/pipeline/delete.ts

```diff
@@ -25,10 +24,9 @@ export function registerPipelineDeleteRoute(router: IRouter) {
       router.handleLegacyErrors(async (context, request, response) => {
         const client = context.logstash!.esClient;
 
-        await client.callAsCurrentUser('delete', {
-          index: INDEX_NAMES.PIPELINES,
-          id: request.params.id,
-          refresh: 'wait_for',
+        await client.callAsCurrentUser('transport.request', {
+          path: '/_logstash/pipeline/' + encodeURIComponent(request.params.id),
```

Comment:
@delvedor I can't find `/_logstash` support in the new Elasticsearch client. Why it so? 
https://github.com/elastic/elasticsearch-js/blob/master/api/kibana.d.ts

Reply:
I'm not aware of that endpoint, and I wasn't able to find it in the [rest-api-spec](https://github.com/elastic/elasticsearch/tree/master/rest-api-spec/src/main/resources/rest-api-spec/api) (and [here](https://github.com/elastic/elasticsearch/tree/master/x-pack/plugin/src/test/resources/rest-api-spec/api). are you sure it's a public API?

Reply:
I can't see them in the spec either, but there present in the code 
https://github.com/elastic/elasticsearch/blob/fb7471b32955c80f1fe850054d68d5b66aca2896/x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/rest/RestGetPipelineAction.java#L32
https://github.com/elastic/elasticsearch/blob/fb7471b32955c80f1fe850054d68d5b66aca2896/x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/rest/RestPutPipelineAction.java#L35
https://github.com/elastic/elasticsearch/blob/fb7471b32955c80f1fe850054d68d5b66aca2896/x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/rest/RestDeletePipelineAction.java#L34

Reply:
The intent for these Logstash APIs is that they are for communication between the stack components and Elasticsearch so we intentionally did not publish a rest api spec for these. If this is necessary, we can probably add specs for these APIs.

Reply:
> communication between the stack components and Elasticsearch so we intentionally did not publish a rest api spec for these.

Can we add a separate REST API spec for such cases? We need to make sure that all the changes in REST API are reflected in the Kibana code as well - we rely on TypeScript type definitions generated from REST API spec.

Reply:
++ another vote on this. I'd be nice to leverage the safety net y'all created with the API spec, but for product to product APIs

Reply:
Sorry for missing an update here, the ES team discussed this and came to the conclusion that we'd add a spec and docs for this API. This work hasn't been started yet.

Reply:
The API spec files have been merged in https://github.com/elastic/elasticsearch/pull/67788